contrib/scoverage/src/ScoverageModule.scala

```diff
@@ -141,8 +173,23 @@ trait ScoverageModule extends ScalaModule { outer: ScalaModule =>
   }
 
   val scoverage: ScoverageData = new ScoverageData(implicitly)
+
   class ScoverageData(ctx0: mill.define.Ctx) extends Module()(ctx0) with ScalaModule {
 
+    /** Coverage in the Scala 3 compilier is only supported in the 3.2.x series and above. */
+    private def isSupportedScala: Task[Boolean] = T.task {
```

Comment:
One thing I'm having a hard time with is figuring out where to put this check. At first I thought it'd be better to put it up above and in every task that computes something I could use like
```scala
    private def ensureValidScala3[A](value: A): Task[A] = T.task {
      val scalaVersion = outer.scalaVersion()
      scalaVersion.split('.') match {
        case Array(major, minor, patch) if major == "3" && minor.toIntOption.exists(_ >= 2) => value
        case Array(major, minor, patch) if major == "3" =>
          val msg =
            s"Detected Scala version: ${scalaVersion}. However, to use Coverage with Scala 3 you must be at least on Scala 3.2.0."
          Result.Failure(msg)
        case _ => value
      }
    }
```
So then I could just do like:
```scala
  def scoverageRuntimeDeps: T[Agg[Dep]] = T {
    if (isScala3()) {
      ensureValidScala3(Agg.empty)()
    } else {
      Agg(ivy"org.scoverage::scalac-scoverage-runtime:${outer.scoverageVersion()}")
    }
  }
```
But that fell apart when I hit `scoveragteToolsClasspath` and couldn't get it to work.

Then I moved this down below into `ScoverageData` and I thought about adding an `assert` in here to check for an invalid Scala 3, but can't really access the scalaVersion in the assert. I'm not really sure where to do this check. Any input on where you'd think this would be best and how to do it?

Reply:
Moreover this should also check that if this is a valid Scala 3 version then the user must _also_ be using scoveage 2.x. It'd be nice to do this check in one place.

Reply:
I think this check is best placed in `scoverageToolsClasspath`. It has access to all required data and is run before any worker action.